optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/__init__.py

```diff
@@ -125,25 +132,45 @@ def handle_control_tag_using(name, using, value):
             if using:
                 raise ValueError("!using can be used only once per "
                                  "node! (%s:%s)" % (path, name))
-            using = value[1]
+            using = value
             if using[0] == '/':
                 using = using[1:]
             if using[-1] == '/':
                 using = using[:-1]
             return using
 
         node = cls_node(str(name))
+        if not values:
+            return node
         using = ''
-        for value in values:
-            if isinstance(value, cls_node):
-                node.add_child(value)
-            elif isinstance(value[0], mux.Control):
-                if value[0].code == YAML_USING:
-                    using = handle_control_tag_using(name, using, value)
+
+        if isinstance(values, dict):    # didn't looked like a node
+            for key, value in values.iteritems():
+                if isinstance(key, mux.Control):
+                    if key.code == YAML_USING:
+                        using = handle_control_tag_using(name, using, value)
+                    else:
+                        handle_control_tag(node, [key, value])
+                elif (isinstance(value, collections.OrderedDict) or
+                      value is None):
+                    node.add_child(tree_node_from_values(key, value))
                 else:
-                    handle_control_tag(node, value)
-            else:
-                node.value[value[0]] = value[1]
+                    node.value[key] = value
+        else:       # already looked like a dict
+            for value in values:
+                if isinstance(value, cls_node):
+                    node.add_child(value)
+                elif isinstance(value[0], mux.Control):
+                    if value[0].code == YAML_USING:
+                        using = handle_control_tag_using(name, using, value[1])
+                    else:
+                        handle_control_tag(node, value)
+                elif isinstance(value[1], collections.OrderedDict):
+                    node.add_child(tree_node_from_values(str(value[0]),
+                                                         value[1]))
+                else:
+                    node.value[value[0]] = value[1]
```

Comment:
Code looks good.. I tested this in some different ways and it seems to be working fine.

Reply:
Running some additional tests